theano/gof/type.py

```diff
@@ -603,26 +633,32 @@ class CDataType(Type):
         The type of the pointer (complete with the `*`).
 
     freefunc
-        A function to call to free the pointer. This function must have a `void`
-        return and take a single pointer argument.
+        A function to call to free the pointer. This function must
+        have a `void` return and take a single pointer argument.
 
     """
 
-    import ctypes
-    if PY3:
-        _cdata_type = ctypes.py_object.from_address(
-            ctypes.addressof(ctypes.pythonapi.PyCapsule_Type)).value
-    else:
-        _cdata_type = ctypes.py_object.from_address(
-            ctypes.addressof(ctypes.pythonapi.PyCObject_Type)).value
-    del ctypes
-
-    def __init__(self, ctype, freefunc=None):
+    def __init__(self, ctype, freefunc=None, headers=None, header_dirs=None,
```

Comment:
I use the same names as the methods you have to define on a type (minus the `c_` prefix).